src/gui/qgsbrowserdockwidget.h

```diff
@@ -62,94 +84,169 @@ class QgsBrowserPropertiesWidget : public QWidget
     virtual void setCondensedMode( bool condensedMode ) { Q_UNUSED( condensedMode ); }
 };
 
-class QgsBrowserLayerProperties : public QgsBrowserPropertiesWidget, private Ui::QgsBrowserLayerPropertiesBase
+/**
+ * \ingroup gui
+ * The QgsBrowserLayerProperties class
+ * \since QGIS 2.10
+ */
+class GUI_EXPORT QgsBrowserLayerProperties : public QgsBrowserPropertiesWidget, private Ui::QgsBrowserLayerPropertiesBase
 {
     Q_OBJECT
   public:
+
+    /**
+      * Constructor for QgsBrowserLayerProperties
+      * \param parent parent widget
+      */
     explicit QgsBrowserLayerProperties( QWidget *parent = nullptr );
+    //! Set item
     void setItem( QgsDataItem *item ) override;
 
+    /** Sets whether the properties widget should display in condensed mode, ie, for display in a dock
+     * widget rather than it's own separate dialog.
+     * \param condensedMode set to true to enable condensed mode
+     * \since QGIS 2.10
+     */
     virtual void setCondensedMode( bool condensedMode ) override;
 
   private:
     QgsBrowserPropertiesWrapLabel *mUriLabel = nullptr;
 };
 
-class QgsBrowserDirectoryProperties : public QgsBrowserPropertiesWidget, private Ui::QgsBrowserDirectoryPropertiesBase
+/**
+ * \ingroup gui
+ * The QgsBrowserDirectoryProperties class
+ * \since QGIS 2.10
+ */
+class GUI_EXPORT QgsBrowserDirectoryProperties : public QgsBrowserPropertiesWidget, private Ui::QgsBrowserDirectoryPropertiesBase
 {
     Q_OBJECT
   public:
+
+    /**
+      * Constructor for QgsBrowserDirectoryProperties
+      * \param parent parent widget
+      */
     explicit QgsBrowserDirectoryProperties( QWidget *parent = nullptr );
 
+    //! Create widget from the given item and add it
     void setItem( QgsDataItem *item ) override;
   private:
     QgsDirectoryParamWidget *mDirectoryWidget = nullptr;
     QgsBrowserPropertiesWrapLabel *mPathLabel = nullptr;
 };
 
-class QgsBrowserPropertiesDialog : public QDialog, private Ui::QgsBrowserPropertiesDialogBase
+/**
+ * \ingroup gui
+ * The QgsBrowserPropertiesDialog class
+ * \since QGIS 2.10
+ */
+class GUI_EXPORT QgsBrowserPropertiesDialog : public QDialog, private Ui::QgsBrowserPropertiesDialogBase
 {
     Q_OBJECT
   public:
+
+    /**
+      * Constructor for QgsBrowserPropertiesDialog
+      * \param settingsSection prefix for settings (from the object name)
+      * \param parent parent widget
+      */
     QgsBrowserPropertiesDialog( const QString &settingsSection, QWidget *parent = nullptr );
     ~QgsBrowserPropertiesDialog();
 
+    //! Create dialog from the given item and add it
     void setItem( QgsDataItem *item );
 
   private:
     QgsBrowserPropertiesWidget *mPropertiesWidget = nullptr;
     QString mSettingsSection;
 };
 
+/**
+ * \ingroup gui
+ * The QgsBrowserDockWidget class
+ * \since QGIS 2.10
```

Comment:
Should read QGIS 3.0 as the class has not been available in API in earlier versions